### ISSUE TITLE
Random "updates" with cron

### ISSUE THREAD

**Issue by vladdlen** [FezVrasta/ark-server-tools#502]
Hi,

I have this added in cron:
*/15 * * * * /usr/local/bin/arkmanager update --warn --saveworld --update-mods

It's been working perfectly up until yesterday evening, when I had 4-5 apparent server updates where the server announced a restart, "updated" and came back online. Today I've had 3 of them. 2 just now the past hour.

The problem is that there's nothing to update. All of the mods are up to date and the server is up to date. Somehow it detects that there's an update while there isn't. The only file edited inside the ARK folder on these occasions is the one called "arkversion".

Any idea as to what may cause this?

**Comment by klightspeed** [comment]
According to https://steamdb.info/app/376030/history/, there was an update about 2 hours ago.

Do any files in `$steamcmdroot/steamapps/workshop/content/346110` have a modified time around that time?

**Comment by arkfrancegaming** [comment]
same problem for me here :
21:48:25: update to 1167275 complete
21:58:14: update to 1167275 complete
22:08:26: update to 1167275 complete
22:18:13: update to 1167275 complete
22:28:15: update to 1167275 complete
22:58:24: update to 1167275 complete
23:08:25: update to 1167275 complete

**Comment by klightspeed** [comment]
The update that was released at 19:38 UTC was 1168781.

It sounds like the update isn't actually completing.

What do you get if you run:
```
arkmanager useconfig pvp update --warn --verbose
```

**Comment by arkfrancegaming** [comment]
steam@ns3013274:/root$ arkmanager useconfig main update --warn --verbose
Running command 'update' for instance 'main'
Le serveur va redemarrer dans 5 minutes pour faire la mise a jour de ARK dans (Reboot for ARK update). Les modifications sont disponibles sur notre site http://arkfrancegaming.com/. IL Y A UNE SAVE ET CELA NE POSE PAS DE PROBLEME POUR LE TAMING, AUGMENTEZ JUSTE LA TORPEUR
Le serveur va redemarrer dans 5 minutes pour faire la mise a jour de ARK dans (Reboot for ARK update). Les modifications sont disponibles sur notre site http://arkfrancegaming.com/. IL Y A UNE SAVE ET CELA NE POSE PAS DE PROBLEME POUR LE TAMING, AUGMENTEZ JUSTE LA TORPEUR
Command processed
There are 4 players connected

what does it should show ?

**Comment by klightspeed** [comment]
It looks like you don't have `arkStagingDir` set.  Therefore it can't download the update until it has stopped the server.

In a minute, it should shut down, and start downloading the update.  Once that has finished, please let us know what it shows.

**Comment by arkfrancegaming** [comment]
it always worked like that, i think the update is the issue...

**Comment by arkfrancegaming** [comment]
steam@ns3013274:/root$ arkmanager useconfig main update --warn --verbose
Running command 'update' for instance 'main'
Copying to staging directory
Downloading ARK updateExecuting /home/steam//steamcmd.sh +@NoPromptForPassword 1 +login anonymous +force_install_dir /home/steam/ARK-Staging +app_update 376030 +quit
Redirecting stderr to '/home/steam/Steam/logs/stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation
-- type 'quit' to exit --
Loading Steam API...Created shared memory when not owner SteamController_Shared_mem
OK.
"@NoPromptForPassword" = "1"

Connecting anonymously to Steam Public...Logged in OK
Waiting for license info...OK
 Update state (0x3) reconfiguring, progress: 0,00 (0 / 0)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
 Update state (0x61) downloading, progress: 79,27 (274934996 / 346819760)
Error! App '376030' state is 0x406 after update job.
Update download interrupted

**Comment by vladdlen** [comment]
klightspeed, thank you for your reply. 

Nothing was changed in `~/steamcmd/steamapps/workshop/content/346110$`. The latest update there was a mode I installed earlier. 

I see that arkfrancegaming had a similar problem. Can it be that arkmanager starts the server without applying the update properly and then detecting the same update again the next time cron runs? I mean, if steam was having issues for example?

The reason for the short intervals between checks is because I want as little downtime as possible. I have been running ARK servers on and off for a few months using arkmanager and I can't recall having this problem before. The cron job has always been set to 15 minutes. 

I stopped the cron job earlier, but when I ran arkmanager update now, it installed with the version number you are referring to: 1168781.
